src/node_jsvmapi.cc

```diff
@@ -1071,6 +1108,21 @@ void* napi_unwrap(napi_env e, napi_value jsObject) {
   return v8impl::ObjectWrapWrapper::Unwrap(jsObject);
 }
 
+int napi_get_internal_field_count(napi_env env, napi_value jsObject) {
+  v8::Local<v8::Object> obj = v8impl::V8LocalValueFromJsValue(jsObject).As<v8::Object>();
+  return obj->InternalFieldCount();
+}
+
+void* napi_get_internal_field_pointer(napi_env env, napi_value jsObject, int index) {
+  v8::Local<v8::Object> obj = v8impl::V8LocalValueFromJsValue(jsObject).As<v8::Object>();
+  return obj->GetAlignedPointerFromInternalField(index);
+}
+
+void napi_set_internal_field_pointer(napi_env env, napi_value jsObject, int index, void* nativeObj) {
+  v8::Local<v8::Object> obj = v8impl::V8LocalValueFromJsValue(jsObject).As<v8::Object>();
+  obj->SetAlignedPointerInInternalField(index, nativeObj);
+}
+
```

Comment:
@boingoing can you confirm that chakra has the equivalent of internal fields, I'm a bit worried this might be v8 specific.